modules/libcom/RTEMS/posix/rtems_init.c

```diff
@@ -0,0 +1,1174 @@
+            char const * new_ntp_servers = "new_ntp_servers";
+            char const * new_host_name = "new_host_name";
+            char const * new_tftp_server_name = "new_tftp_server_name";
+            
+            if (!strncmp(name, new_ntp_servers, strlen(new_ntp_servers)))
+                snprintf(rtemsInit_NTP_server_ip,
+                         sizeof(rtemsInit_NTP_server_ip),
+                         "%s", value);
+            
+            if (!strncmp(name, new_host_name, strlen(new_host_name)))
+                sethostname (value, strlen (value));
+            
+            if (!strncmp(name, new_tftp_server_name, strlen(new_tftp_server_name))){
+                snprintf(rtems_bsdnet_bootp_server_name,
+                         sizeof(bootp_server_name_init),
+                         "%s", value);
+                printf(" rtems_bsdnet_bootp_server_name : %s\n", rtems_bsdnet_bootp_server_name);
+            }
+            if(!strncmp(name, "new_bootfile_name", 20)){
+                snprintf(rtems_bsdnet_bootp_boot_file_name,
+                         sizeof(bootp_boot_file_name_init),
+                         "%s", value);
+                printf(" rtems_bsdnet_bootp_boot_file_name : %s\n", rtems_bsdnet_bootp_boot_file_name);
+            }
+            if(!strncmp(name, "new_rtems_cmdline", 20)){
+                snprintf(rtems_bsdnet_bootp_cmdline,
+                         sizeof(bootp_cmdline_init),
+                         "%s", value);
+                printf(" rtems_bsdnet_bootp_cmdline : %s\n", rtems_bsdnet_bootp_cmdline);
+            }
+          }
+        ++env;
+    }
+    if (bound)
+        epicsEventSignal(dhcpDone);
+}
+
+static rtems_dhcpcd_hook dhcpcd_hook = {
+	.name = "ioc boot",
+	.handler = dhcpcd_hook_handler
+};
+
+static void
+default_network_on_exit(int exit_code, void *arg)
+{
+        rtems_printer printer;
+
+        (void)arg;
+
+        rtems_print_printer_printf(&printer);
+        rtems_stack_checker_report_usage_with_plugin(&printer);
+}
+
+static void
+default_network_set_self_prio(rtems_task_priority prio)
+{
+        rtems_status_code sc;
+
+        sc = rtems_task_set_priority(RTEMS_SELF, prio, &prio);
+        assert(sc == RTEMS_SUCCESSFUL);
+}
+
+static void
+default_network_dhcpcd(void)
+{
+    static const char default_cfg[] = "clientid FHI test client\n";
+    rtems_status_code sc;
+    int fd;
+    int rv;
+    ssize_t n;
+
+    fd = open("/etc/dhcpcd.conf", O_CREAT | O_WRONLY,
+        S_IRWXU | S_IRWXG | S_IRWXO);
+    assert(fd >= 0);
+
+    n = write(fd, default_cfg, sizeof(default_cfg) - 1);
+    assert(n == (ssize_t) sizeof(default_cfg) - 1);
+
+    static const char fhi_cfg[] =
+        "nodhcp6\n" \
+        "ipv4only\n" \
+        "option ntp_servers\n" \
+        "option rtems_cmdline\n" \
+        "option tftp_server_name\n" \
+        "option bootfile_name\n" \
+        "define 129 string rtems_cmdline\n";
```

Comment:
Probably should add `timeout 0` to disable DHCP timeout (imo. a must for a production IOC).  Could also add `noipv6`.

I see `waitip 4` which, when desirable, seems like it would remove the need for a custom hook in `rtems_init.c`.

Also, for testing, it looks like you could set something like:

```
interface eth0
static ip_address=192.168.0.10/24
static routers=192.168.0.1
static domain_name_servers=192.168.0.1
```

cf. `man dhcpcd/dhcpcd.conf.5.in` from the libbsd source.

Reply:
Ah, right `ipv4only` and `noipv6` are practically equivalent.